src/modules/pair/services/pair.metadata.builder.ts

```diff
@@ -36,126 +57,266 @@ export class PairsMetadataBuilder {
     }
 
     async filterByIssuedLpToken(): Promise<PairsMetadataBuilder> {
-        this.pairsMetadata = await this.filteringService.pairsByIssuedLpToken(
+        await this.indexLpTokens();
+
+        this.pairs = PairFilteringService.pairsByIssuedLpToken(
             this.filters,
-            this.pairsMetadata,
+            this.pairs,
         );
         return this;
```

Comment:
All these filters from `PairsMetadataBuilder` should return the actual filtered pairs, not an instance of the class